analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -83,14 +83,23 @@ abstract class PackageManager {
                     val filesInDir = dir.toFile().listFiles()
 
                     packageManagers.forEach { manager ->
-                        val matches = manager.matchersForDefinitionFiles.mapNotNull { glob ->
-                            filesInDir.find { file ->
-                                glob.matches(file.toPath())
+                        // Create a list of lists of matching files per glob.
+                        val matchesPerGlob = manager.matchersForDefinitionFiles.mapNotNull { glob ->
+                            // Create a list of files in the current directory that match the current glob.
+                            val filesMatchingGlob = filesInDir.filter { file ->
+                                file != null && glob.matches(file.toPath())
                             }
+                            filesMatchingGlob.takeIf { it.isNotEmpty() }
                         }
 
-                        if (matches.isNotEmpty()) {
-                            result.getOrPut(manager) { mutableListOf() }.add(matches.first())
+                        if (matchesPerGlob.isNotEmpty()) {
+                            // Only consider all matches for the first glob that has matches. This is because globs are
+                            // defined in order of priority, and multiple globs may just be alternative ways to detect
+                            // the exact same project.
+                            // That is, at the example of a PIP project, if a directory contains all three files
+                            // "requirements-py2.txt", "requirements-py3.txt" and "setup.py", only consider the
+                            // former two as they match the glob with the highest priority, but ignore "setup.py".
+                            result.getOrPut(manager) { mutableListOf() }.addAll(matchesPerGlob.first())
```

Comment:
This probably also requires an update of the documentation for `@param globsForDefinitionFiles` in `PackageManagerFactory`.

Reply:
Not really, as the docs were generic enough :wink: But I added a sentence to document the current behavior in more detail.